airbyte-cdk/java/airbyte-cdk/db-sources/src/testFixtures/java/io/airbyte/cdk/integrations/standardtest/source/DataSourceFactoryTest.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.integrations.standardtest.source;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import com.zaxxer.hikari.HikariDataSource;
+import io.airbyte.cdk.db.factory.DataSourceFactory;
+import java.util.Map;
+import javax.sql.DataSource;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+abstract public class DataSourceFactoryTest<T extends JdbcDatabaseContainer<T>> {
+
+  protected void testCreatingDataSourceWithConnectionTimeoutSetBelowDefault(T container, int defaultTimeout, String timeoutKey, int explicitTimeout) {
```

Comment:
I don't think this is particularly elegant, but neither is the similar code in `DataSourceBuilder.getConnectionTimeoutMs :(`

Reply:
Agreed, but this doesn't need to be an abstract class.